lib/core/test/unit/Cardano/Wallet/DB/Arbitrary.hs

```diff
@@ -671,5 +736,27 @@ instance Buildable (PrimaryKey WalletId) where
 instance Buildable MockChain where
     build (MockChain chain) = blockListF' mempty build chain
 
-instance Arbitrary SomeMnemonic where
-    arbitrary = SomeMnemonic <$> genMnemonic @12
+instance Buildable (SharedState 'Mainnet ShelleyKey) where
+    build (PendingSharedState prefix accXPub pTemplate dTemplateM g) =
+        build (   printStatePending
+               <> printIndex prefix
+               <> printAccXPub
+               <> printPaymentScript
+               <> printDelegationScript
+               <> printGap
+              )
+      where
+        printIndex (DerivationPrefix (_,_,ix)) = " hardened index: "<> toText (getIndex ix)
+        printStatePending = "shared wallet state: pending"
+        printPaymentScript = " payment script template: "<> T.pack (show pTemplate)
+        printDelegationScript = " delegation script template: " <> case dTemplateM of
+            Nothing -> "absent"
+            Just dTemplate -> T.pack (show dTemplate)
+        printAccXPub = " accXPub: " <> T.pack (show accXPub)
+        printGap = " gap: " <> toText (Seq.getAddressPoolGap g)
+    build (SharedState prefix pool) =
+        build (printStateActive <> printIndex prefix) <> printPool
+      where
+        printStateActive = "shared wallet state: active"
+        printPool = build pool
+        printIndex (DerivationPrefix (_,_,ix)) = " hardened index: "<> toText (getIndex ix)
```

Comment:
Do you have an example of what this would produce? It appears to me that this may produce an quite long and maybe not really readable line :thinking:  ?

Reply:
Ping :no_mouth: ?

Reply:
in the end we use ReadyFields (standing for active wallet) in DB tests, mainly to deal with static parts fix, made some cleaning, now the example `pretty state` looks like:
```
shared wallet state: active hardened index: 2147483648 multisig_script (ParentContext : 0730f885...246cd32e,  payment script credential: Cosigners:| cosigner#0 0730f885...246cd32e Script:RequireAllOf [RequireSignatureOf (Cosigner 0),RequireAnyOf [ActiveFromSlot 200,ActiveUntilSlot 100]] delegation script credential: absent) (gap=10)
```